File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/abstract_factory/AccessUser.java
package jv.com.tencent.cain.design_pattern.abstract_factory;

/**
 * @author : jiangyu
 * @date : 2020/3/20
 * @desc : xxx
 */
public class AccessUser implements IUser {
    @Override
    public void insertUser(User user) {
        System.out.println("access# insert user: " + user);
    }

    @Override
    public User queryUser(int id) {
        return new User("Richard", 30);
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/abstract_factory/AccessUser.kt
package kt.com.tencent.cain.desigin_pattern.abstract_factory

/**
 * @author : jiangyu
 * @date   : 2020/3/20
 * @desc   : xxx
 */
class AccessUser : IUser {
    override fun insertUser(user: User) {
        println("access# insert user: $user")
    }

    override fun queryUser(id: Int): User = User("Richard", 30)
}